frontend/src/metabase-types/api/settings.ts

```diff
@@ -353,3 +363,48 @@ export type Settings = InstanceSettings &
 export type SettingKey = keyof Settings;
 
 export type SettingValue = Settings[SettingKey];
+
+// TODO: Keep enterprise settings types away from other settings types
+export type SettingElementKey = EnterpriseSettingKey;
+
+// The type for settings within the Admin sections.
+// Used for displaying forms, submitting changes, etc.
+export type SettingElement<Key extends SettingElementKey> =
+  SettingDefinition<Key> & {
+    tab?: string;
+    display_name?: string;
+    type?: string;
+    description?: string | ReactNode;
+    note?: string;
+    searchProp?: string;
+    placeholder?: string;
+    options?: { value: EnterpriseSettings[Key]; name: string }[];
+    originalValue?: EnterpriseSettings[Key];
+    defaultValue?: EnterpriseSettings[Key];
+    required?: boolean;
+    autoFocus?: boolean;
+    showActualValue?: boolean;
+    allowValueCollection?: boolean;
+    noHeader?: boolean;
+    disableDefaultUpdate?: boolean;
+    validations?: [string, string][];
+    widget?: ComponentType<any>;
+    warningMessage?: string;
+    postUpdateActions?: VoidFunction[];
+    getProps?: (setting: SettingDefinition<Key>) => Record<string, any>;
+    getHidden?: (
+      settingValues: EnterpriseSettings,
+      derivedSettingValues: EnterpriseSettings,
+    ) => boolean;
+    onChanged?: (
+      oldValue: string,
+      newValue: string,
+      settingsValues: EnterpriseSettings,
+      handleChangeSetting: (key: string, value: string) => void,
+    ) => void;
+    onBeforeChanged?: (oldValue: string, newValue: string) => void;
+  };
```

Comment:
Ok, makes sense. I had the idea of wanting to keep all of the settings things together (I don't think any of us knew that this type existed since it was kinda hanging out in a folder), but I can move it back